setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php

```diff
@@ -87,6 +87,10 @@ public function getList($path)
      */
     private function isExclude(\SplFileInfo $fileItem, $patterns)
     {
+        if($fileItem->getBasename()[0] == '.') {
```

Comment:
If you add this check here, it will be invoked for every pattern, while it need to be invoked just once per file. The better place for this piece will be here as a third skipped case

```
if ($fileItem->isDir() || $fileItem->getExtension() !== 'php') {
                continue;
            }
```